php-plugin/src/test/resources/checks/AvoidGettingSizeCollectionInForLoop.php

```diff
@@ -1,79 +1,147 @@
 <?php
 
-$food = array('orange', 'banana', 'apple', 'carrot', 'collard', 'pea');
+$array = array('orange', 'banana', 'apple', 'carrot', 'collard', 'pea');
 
 /**
- * FOR STATEMENTS
+ * FOR STATEMENTS // RIGHT OPERAND
  */
- for ($i = 0; $i < count($food); $i++) {  // NOK {{Avoid getting the size of the collection in the loop}}
-     var_dump($food[$i]);
- }
+for ($i = 0; $i < count($array); ++$i) {  // NOK {{Avoid getting the size of the collection in the loop}}
+    var_dump($array[$i]);
+}
+
+$size = count($array);
+for ($i = 0; $i < $size; ++$i) {  // Compliant
+    var_dump($array[$i]);
+}
+
+for ($i = 0; $i < sizeof($array); ++$i) {  // NOK {{Avoid getting the size of the collection in the loop}}
+    var_dump($array[$i]);
+}
 
- $size = sizeof($food);
- for ($i = 0; $i < $size; $i++) {  // Compliant
-     var_dump($food[$i]);
- }
+$size = sizeof($array);
+for ($i = 0; $i < $size; ++$i) {  // Compliant
+    var_dump($array[$i]);
+}
 
- for ($i = 0; $i < sizeof($food); $i++) {  // NOK {{Avoid getting the size of the collection in the loop}}
-     var_dump($food[$i]);
- }
+for ($i = 0; $i < iterator_count(new ArrayIterator($array)); ++$i) {  // NOK {{Avoid getting the size of the collection in the loop}}
+    var_dump($array[$i]);
+}
 
- $size = count($food);
- for ($i = 0; $i < $size; $i++) {  // Compliant
-     var_dump($food[$i]);
- }
+$size = iterator_count(new ArrayIterator($array));
+for ($i = 0; $i < $size; ++$i) {  // Compliant
+    var_dump($array[$i]);
+}
+
+/**
+ * FOR STATEMENTS // LEFT OPERAND
+ */
+for ($i = 0; count($array) > $i; ++$i) {  // NOK {{Avoid getting the size of the collection in the loop}}
+    var_dump($array[$i]);
+}
+
+$size = count($array);
+for ($i = 0; $size > $i; ++$i) {  // Compliant
+    var_dump($array[$i]);
+}
+
+for ($i = 0; sizeof($array) > $i; ++$i) {  // NOK {{Avoid getting the size of the collection in the loop}}
+    var_dump($array[$i]);
+}
+
+$size = sizeof($array);
+for ($i = 0; $size > $i; ++$i) {  // Compliant
+    var_dump($array[$i]);
+}
+
+for ($i = 0; iterator_count(new ArrayIterator($array)) > $i; ++$i) {  // NOK {{Avoid getting the size of the collection in the loop}}
+    var_dump($array[$i]);
+}
+
+$size = iterator_count(new ArrayIterator($array));
+for ($i = 0; $size > $i; ++$i) { // Compliant
+    var_dump($array[$i]);
+}
 
 /**
  * WHILE STATEMENTS
  */
 $i = 0;
-while($i < count($food)) // NOK {{Avoid getting the size of the collection in the loop}}
-{
-    var_dump($food[$i]);
-    $i++;
+while ($i < count($array)) { // NOK {{Avoid getting the size of the collection in the loop}}
+    var_dump($array[$i]);
+    ++$i;
+}
+
+$i = 0;
+$size = count($array);
+while ($i < $size) { // Compliant
+    var_dump($array[$i]);
+    ++$i;
+}
+
+$i = 0;
+while ($i < sizeof($array)) { // NOK {{Avoid getting the size of the collection in the loop}}
```

Comment:
maybe here, put sizeof call in left operator to have two different tests with while loop (one with right operator above, and one with left operator here)

Reply:
I don’t see the use case with a sizeof as the left operator? But yes I think the rule should check the presence of sizeof no matter its implacement in the `while ([...])` (that seems to be good based on the rule code)